### ISSUE TITLE
[Re] Context-Dependent Encoding of Fear and Extinction Memories in a Large-Scale Network Model of the Basal Amygdala

### ISSUE THREAD

**Issue by tawantayron** [ReScience/submissions#49]
**Original article:** Vlachos, I., Herry, C., Lüthi, A., Aertsen, A., & Kumar, A. "[Context-dependent encoding of fear and extinction memories in a large-scale network model of the basal amygdala](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1001104)". PLoS Comput Biol, v. 7, n. 3, p. e1001104, 2011.

**PDF URL:** https://github.com/tawantayron/RE-Fear-and-extinction-in-a-BA-net-model/blob/master/article.pdf
**Metadata URL:** https://github.com/tawantayron/RE-Fear-and-extinction-in-a-BA-net-model/blob/master/metadata.yaml
**Code URL:** https://github.com/tawantayron/RE-Fear-and-extinction-in-a-BA-net-model

**Scientific domain:** Computational Neuroscience
**Programming language:** Python and Brian 2
**Suggested editor:** @benoit-girard
**Suggested reviewer:** @mstimberg @heplesser @rgutzen @TiinaManninen

**Comment by rougier** [comment]
@tawantayron Thanks for your submission, we'll assign an editor soon.

**Comment by rougier** [comment]
@benoit-girard @otizonaizit @oliviaguest @eroesch @gdetor Can one of you edit this submission?

**Comment by eroesch** [comment]
Yes, I can do it.

**Comment by rougier** [comment]
Thanks. I've just assigned you.

**Comment by tawantayron** [comment]
Dear editor,
We would like to kindly ask if there is any update on the progress of the review process of our submission.
We understand that we still are amidst difficult times and that the steps involving the revision of our manuscript can be longer than usual, but it would be great if this manuscript could be progressed in the not too distant future.
Cordially,

**Comment by eroesch** [comment]
Dear @tawantayron, I am working on it. It has been a bit difficult to find reviewers, but I remain optimistic. I will provide you with my own comments very soon.

**Comment by eroesch** [comment]
Hi @tawantayron, I am running into a runtime error, running runsim.py with protocol=1, =2 (I am still running 3 and above, some warnings but so far so good). See below. The code seems to carry on in a loop, and I assumed the code might still be running. After 24h+ of running though, it still does not complete. Can you let me know what you think?

ERROR      Brian 2 encountered an unexpected error. If you think this is a bug in Brian 2, please report this issue either to the discourse forum at <http://brian.discourse.group/>, or to the issue tracker at <https://github.com/brian-team/brian2/issues>. Please include this file with debug information in your report: /var/folders/8f/5f00v5l94r91mtfll9t0x2sm0000gn/T/brian_debug_anijad72.log  Additionally, you can also include a copy of the script that was run, available at: /var/folders/8f/5f00v5l94r91mtfll9t0x2sm0000gn/T/brian_script_54_70__1.py Thanks! [brian2]
Traceback ....

RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.

**Comment by eroesch** [comment]
Actually I spoke to soon, protocol=3 works, but not 4 or 5.

**Comment by tawantayron** [comment]
Dear editor,

We have double-checked the code on Linux systems and fix small issues encountered. We also noted a bug when running on MacOS that should be solved now, in case that is the system you are using let us know if you run into more issues while executing the code.

**Comment by eroesch** [comment]
@tawantayron, awesome. Thanks. Yes, MacOS. I will pull and try again.
What was the issue?

**Comment by niltonlk** [comment]
Dear @eroesch,
The issue was that in MacOS, the code block that creates the pool (`p = multiprocessing.Pool(processes=n_procs)`) and calls `p.map(func, arg_list)` to run some simulations in parallel (with different parameters), should be enclosed in an if clause (`if __name__ == '__main__':`) to avoit that the code is called by every process (codes that reads the simulation data and plots the results were also included withing the same if clause). This was not necessary on Linux, which is the environment we have originally tested.

**Comment by tawantayron** [comment]
Dear editor @eroesch , we kindly ask for any update on the review process.
Cordially,

**Comment by rougier** [comment]
@eroesch Gentle reminder (but maybe on vacation).

**Comment by tawantayron** [comment]
Dear editor @eroesch , we kindly ask again for an update. There was no feedback since our last correction. Did you manage to run the codes normally or is there still something missing?

**Comment by rougier** [comment]
@eroesch Any update ? Also, maybe @benoit-girard would be interested in reviewing this submission on a Large-Scale Network Model of the Basal Amygdala

**Comment by eroesch** [comment]
Hi! I was indeed off the grid for a while! I had issues running the original code, and haven't had time to try all of @niltonlk's fixes. I have timeblocked time to run through it this week. @benoit-girard, Would you mind giving it a go as well? What would be a good timeline for you?

**Comment by rougier** [comment]
@eroesch @benoit-girard Any update?

**Comment by eroesch** [comment]
Jep. All done. It took a few days to run the last models, sorry. @benoit-girard, I do not think I need you to do a review.

Thank you for fixing the code, I can now confirm that I have managed to run all your scripts, and have reproduced all the figures presented in your submission; the more complex models took a few days. The figures you produce are similar to those in the original work (but not exactly the same), and the equations you have had to implement are close enough to the original ones. I am happy to recommend acceptance. I do have some minor recommendations. Please let me know if you are able to do this; if not I can make some edits myself. @rougier, I am going ahead with the publishing workflow. @tawantayron, do let me know how long you think you'd need.

TODO:
-  Please provide a requirement.txt file containing the required libraries and packages, and their versions that need to be used to run the models. Your README.md is quite verbose, and it would ease the workflow for someone else to reproduce your environment.
- You do a great job describing the missing parameters and the guesswork you have had to do, but you do not follow strictly the way to present the equations from the original paper, which can be confusing for someone who is not from the field. At first glance and for the untrained eye, they do look quite different: particularly, your equations 1-3 and the original 1-2, and your 5 to the original 5. At a minimum, add sentences pointing out the differences and explaining why they are equivalent (e.g. mentioning the equivalence of notation, and explaining the inclusion of the conductance in Eq 5). I do not necessarily think you should rename your variables, but it would help.
- Make sure the captions for all your figures mention the original figure it refers to. Obviously, in the core text, explain any differences. e.g. Your figure 7 mentions the original Figure 5, but really refers to the original Figure 7, showing the renewal scenario. That is not clear.

**Comment by tawantayron** [comment]
Dear Editor @eroesch , we are grateful for your comments and for recommending acceptance of our work. In a few days, we will submit the new version correcting the points you mentioned. Cordially.

**Comment by eroesch** [comment]
Great, thanks @tawantayron. Once you are done, I will edit your files, and will ask you to compile the .tex file again, to include metadata. I will for your repo and produce a pull request for you to review.

**Comment by tawantayron** [comment]
Dear editor @eroesch , we made the following modifications:

- We added the file “requirements.txt” which contains all versions of the packages used to run the models, and also added simplified commands to install an anaconda environment with all the packages needed to run the codes. We also create a document with the instructions to install the packages using pip.
- Thanks to the editor for highlighting this point. We have added brief explanations to the manuscript in order to clarify that the dynamics equations of the models as we wrote are equivalent to the equations used in the original article.
- We reviewed all the figures in the manuscript, clarifying which ones correspond to the reference article and which ones are additional figures that add details to the studied models.

We are at your disposal for any other queries. Cordially,

**Comment by tawantayron** [comment]
Dear @eroesch ,

We have already performed the update of the metadata file and the new PDF version of the manuscript is available. If more changes are still needed, we are at your disposal. Cordially,

**Comment by rougier** [comment]
@eroesch Gentle reminder.

**Comment by eroesch** [comment]
All done. I think: [https://zenodo.org/record/5657320#.YYp2_r2mOAk](https://zenodo.org/record/5657320#.YYp2_r2mOAk).
I am not quite sure I understand how to update the website though, @rougier...

Also, I have had to update the metadata myself and latex the pdf (first because I needed to add the publication date, and then because I have had to re-reserve a doi), then I also hit a bump with the sandbox in step 2 (probably because I wasn't using the appropriate version of PyYAML). All sorted in the end, with a bit of manual sweat.
I am closing the issue.
